core-metadata/src/main/java/org/apache/kylin/metadata/filter/CompareTupleFilter.java

```diff
@@ -74,6 +77,9 @@ private CompareTupleFilter(CompareTupleFilter another) {
 
     @Override
     public void addChild(TupleFilter child) {
+        if (child instanceof CompareTupleFilter) {
```

Comment:
plz add some test case in UT or IT

Reply:
We need to refine our IT and add more test cases. To test this case, it's necessary to add a column with boolean type.